templates/export/resources.js

```diff
@@ -89,13 +89,88 @@ module.exports = Object.assign(
         "Principal": "apigateway.amazonaws.com"
       }
     },
+    "GenesysCodeVersion": {
+      "Type": "Custom::S3Version",
+      "Properties": {
+        "ServiceToken": {"Ref": "CFNLambda"},
+        "Bucket": {"Ref": "BootstrapBucket"},
+        "Key": {"Fn::Sub": "${BootstrapPrefix}/lambda/genesys.zip"},
+        "BuildDate": (new Date()).toISOString()
+      }
+    },
+    "GenesysLambda": {
+      "Type": "AWS::Lambda::Function",
+      "Properties": {
+        "Code": {
+          "S3Bucket": {"Ref": "BootstrapBucket"},
+          "S3Key": {"Fn::Sub": "${BootstrapPrefix}/lambda/genesys.zip"},
+          "S3ObjectVersion": {"Ref": "GenesysCodeVersion"}
+        },
+        "Environment": {
+          "Variables": {
+            outputBucket: {"Ref": "ExportBucket"},
+            s3Prefix: "connect/",
+            accountId: {"Ref": "AWS::AccountId"},
+            region: {"Ref": "AWS::Region"},
+            LexVersion: {"Ref": "LexVersion"},
+            // Lex V1
+            fallBackIntent: {"Ref": "FallbackIntent"},
+            intent: {"Ref": "Intent"},
+            lexBot: {"Ref": "BotName"},
+            // Lex V2
+            LexV2BotName: {"Ref": "LexV2BotName"},
+            LexV2BotId: {"Ref": "LexV2BotId"},
+            LexV2BotAlias: {"Ref": "LexV2BotAlias"},
+            LexV2BotAliasId: {"Ref": "LexV2BotAliasId"},
+            LexV2BotLocaleIds: {"Ref": "LexV2BotLocaleIds"}
+          }
+        },
+        "Handler": "index.handler",
+        "MemorySize": "1024",
+        "Role": {"Fn::GetAtt": ["ExportRole", "Arn"]},
```

Comment:
- Should we share the ExportRole for both ConnectLambda and GenesysLambda or separate Role to scope down each lambda? One reason may be to restrict access to specific S3 paths based on Connect or Lambda for export bucket.
- The ExportPolicy has a s3:* which we can scope down. The resource is scoped down to the bucket.

Reply:
I dove deeper into what is using the ExportPolicy and ExportRole besides the Genesys and Connect exports. There is an 'exports.zip' that contains lambdas for exporting and importing questions-- and it uses a bunch of different s3 functions.  Per Mohsen and Bob P's comments where they tried to separate out functions and broke QnABot, I think we should also leave it as s3:* for now-- will add backlog item to investigate and test this further in next release?